### DISCUSSION TITLE
Java11 - Automatic Instrumentation: Report method name or exact location where trace was taken

### DISCUSSION THREAD

**Discussion by julian-sotec** [open-telemetry/opentelemetry-java-instrumentation#9840] [Q&A] [answered]
Hello otel community,

This is basically a question on understanding the boundaries of this great tracing framework.
I am using open telemetry with java11 on Google App Engine. The libraries Automatic Instrumentation without writing any code
works great in general. Here is how we have set it up:
Using a maven profile we add the open telemetry java libraries with autoconfigure extension
```
        <dependency>
          <groupId>io.opentelemetry</groupId>
          <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
          <version>1.25.0-alpha</version>
        </dependency>
```
And using an custom entrypoint we specify the auto exporter jar:
```
    -Dotel.javaagent.experimental.extensions=/workspace/WEB-INF/otelagent/google-exporter-auto.jar
    -Dotel.traces.exporter=google_cloud_trace
    -Dotel.metrics.exporter=google_cloud_monitoring
    -Dotel.resource.attributes=service.name=my-service,service.namespace=my.namespace
```
With this setup we are able to see the rpc call and also the http request to external api's using client libraries.
Example:

<html>
<body>

key | value
-- | --
/http/method | POST
/http/status_code | 200
/http/url | https://datastore.googleapis.com/v1/projects/my-project:runQuery
/tid | 29
cloud.availability_zone | zone
cloud.region | region
faas.id | somelongid
faas.name | my.servicename
faas.version | my.version
g.co/agent | opentelemetry-java 1.25.0; google-cloud-trace-exporter 0.1.0
g.co/r/gae_instance/instance_id | somelongid
g.co/r/gae_instance/location | region
g.co/r/gae_instance/module_id | my.servicename
g.co/r/gae_instance/version_id | my.version
host.arch | amd64
host.name | localhost
net.peer.name | datastore.googleapis.com
net.transport | ip_tcp
os.description | Linux 4.4.0
otel.scope.name | io.opentelemetry.google-http-client-1.19
otel.scope.version | 1.25.1-alpha
process.pid | 11
process.runtime.description | Ubuntu OpenJDK 64-Bit Server VM 11.0.19+7-post-Ubuntu-0ubuntu118.04.1
process.runtime.version | 11.0.19+7-post-Ubuntu-0ubuntu118.04.1
service.namespace | my.namespace
telemetry.auto.version | 1.25.1
telemetry.sdk.language | java
telemetry.sdk.name | opentelemetry
telemetry.sdk.version | 1.25.0
thread.name | Request

</body>
</html>

However I would like to also see the java method name where the trace was recorded. I guess I know the application
well enough being able to identify the code file - however It would be also beneficial knowing the exact location in code.
Is this possible using automatic instrumentation with some attribute / property setting?
If not - what would be necessary to do to achive this?

Any input is much appriciated.

Cheers.

**Comment by trask** [comment, marked as answer]
hi @julian-sotec! there is no option (currently at least) to capture the caller. this would require getting the stack trace which is a relatively expensive operation and so would probably only be useful for debugging (or under very low sampling rates)

**Reply by julian-sotec** [reply]
Thanks @trask  for pointing this out!
